ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/operations/AggregateAndProof.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.datastructures.operations;
+
+import com.google.common.primitives.UnsignedLong;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.ssz.SSZ;
+import tech.pegasys.artemis.datastructures.util.SimpleOffsetSerializer;
+import tech.pegasys.artemis.util.SSZTypes.SSZContainer;
+import tech.pegasys.artemis.util.bls.BLSSignature;
+import tech.pegasys.artemis.util.sos.SimpleOffsetSerializable;
+
+public class AggregateAndProof implements SimpleOffsetSerializable, SSZContainer {
+
+  // The number of SimpleSerialize basic types in this SSZ Container/POJO.
+  public static final int SSZ_FIELD_COUNT = 1;
+
+  private UnsignedLong index;
+  private BLSSignature selection_proof;
+  private Attestation aggregate;
+
+  public AggregateAndProof(
+      UnsignedLong index, BLSSignature selection_proof, Attestation aggregate) {
+    this.index = index;
+    this.selection_proof = selection_proof;
+    this.aggregate = aggregate;
+  }
+
+  public AggregateAndProof() {
```

Comment:
Do we need this empty constructor + setters?  Or can we get away with an immutable object?

Reply:
I think we can. Let me make that change real quickly.